src/components/Calculator.js

```diff
@@ -1,13 +1,41 @@
+import { useState, useEffect } from 'react';
 import Button from './Button';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
 const buttons = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
-const buttonItem = buttons.map((label) => <Button key={label} labelTag={label} />);
-const Calculator = () => (
-  <div className="calculator">
-    <div className="result">0</div>
-    {buttonItem}
-  </div>
-);
+
+const Calculator = () => {
+  const [calculatorObject, setCalculatorObject] = useState({
+    total: null,
+    next: '0',
+    operation: null,
+  });
+
+  const onClickHandler = (buttonName) => {
+    let result = calculate(calculatorObject, buttonName);
+    if (result.next === null && result.total === null) result = { ...result, next: '0' };
+    setCalculatorObject(result);
+  };
+
+  useEffect(() => {}, [calculatorObject]);
+
```

Comment:
- [ ] [OPTIONAL] In the provided code snippet, the `useEffect` hook doesn't have any effect or purpose, and it can be safely removed without affecting the functionality of the `Calculator` component. Here's why:

The `useEffect` hook is used to perform side effects in a React functional component. It takes two arguments: a function to run, and a dependency array that specifies when the effect should be re-run. In this case, the provided dependency array `[calculatorObject]` indicates that the effect should be re-run whenever the `calculatorObject` state changes.

However, the effect itself is an empty function body: `() => {}`. This means that the effect doesn't perform any actual logic or side effect. It's effectively a no-op (no operation). 😎